### ISSUE TITLE
学习率始终不变

### ISSUE THREAD

**Issue by juemifuji** [baichuan-inc/Baichuan2#287]
hostfile="/path/to/hostfile"
deepspeed --hostfile=$hostfile fine-tune.py  \
    --report_to "none" \
    --data_path "data/belle_chat_ramdon_10k.json" \
    --model_name_or_path "baichuan-inc/Baichuan2-7B-Base" \
    --output_dir "output" \
    --model_max_length 512 \
    --num_train_epochs 4 \
    --per_device_train_batch_size 16 \
    --gradient_accumulation_steps 1 \
    --save_strategy epoch \
    --learning_rate 1e-5 \
    --lr_scheduler_type cosine \
    --adam_beta1 0.9 \
    --adam_beta2 0.98 \
    --adam_epsilon 1e-8 \
    --max_grad_norm 1.0 \
    --weight_decay 1e-4 \
    --warmup_ratio 0.0 \
    --logging_steps 1 \
    --gradient_checkpointing True \
    --deepspeed ds_config.json \
    --bf16 True \
    --tf32 True

将学习率改为  --lr_scheduler_type cosine ，但是训练过程中**学习率始终不变**，试了linear也是如此

22.1.171.4: {'loss': 0.8735, 'learning_rate': 1e-05, 'epoch': 2.94}
22.1.175.3: {'loss': 0.8735, 'learning_rate': 1e-05, 'epoch': 2.94}
22.1.170.153: {'loss': 0.8735, 'learning_rate': 1e-05, 'epoch': 2.94}
22.1.170.153: {'loss': 0.8862, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.171.4: {'loss': 0.8862, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.175.3: {'loss': 0.8862, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.160.119: {'loss': 0.8862, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.171.4: {'loss': 0.8039, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.160.119: {'loss': 0.8039, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.175.3: {'loss': 0.8039, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.170.153: {'loss': 0.8039, 'learning_rate': 1e-05, 'epoch': 2.95}
22.1.160.119: {'loss': 0.777, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.171.4: {'loss': 0.777, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.175.3: {'loss': 0.777, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.170.153: {'loss': 0.777, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.171.4: {'loss': 0.8601, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.160.119: {'loss': 0.8601, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.175.3: {'loss': 0.8601, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.170.153: {'loss': 0.8601, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.171.4: {'loss': 0.8096, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.175.3: {'loss': 0.8096, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.160.119: {'loss': 0.8096, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.170.153: {'loss': 0.8096, 'learning_rate': 1e-05, 'epoch': 2.96}
22.1.160.119: {'loss': 0.8777, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.171.4: {'loss': 0.8777, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.175.3: {'loss': 0.8777, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.170.153: {'loss': 0.8777, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.171.4: {'loss': 0.8917, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.160.119: {'loss': 0.8917, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.170.153: {'loss': 0.8917, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.175.3: {'loss': 0.8917, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.171.4: {'loss': 0.8389, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.170.153: {'loss': 0.8389, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.175.3: {'loss': 0.8389, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.160.119: {'loss': 0.8389, 'learning_rate': 1e-05, 'epoch': 2.97}
22.1.171.4: {'loss': 0.8438, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.175.3: {'loss': 0.8438, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.160.119: {'loss': 0.8438, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.170.153: {'loss': 0.8438, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.175.3: {'loss': 0.8472, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.171.4: {'loss': 0.8472, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.160.119: {'loss': 0.8472, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.170.153: {'loss': 0.8472, 'learning_rate': 1e-05, 'epoch': 2.98}
22.1.171.4: {'loss': 0.8032, 'learning_rate': 1e-05, 'epoch': 2.99}
22.1.175.3: {'loss': 0.8032, 'learning_rate': 1e-05, 'epoch': 2.99}
22.1.160.119: {'loss': 0.8032, 'learning_rate': 1e-05, 'epoch': 2.99}
22.1.170.153: {'loss': 0.8032, 'learning_rate': 1e-05, 'epoch': 2.99}
22.1.160.119: {'loss': 0.8561, 'learning_rate': 1e-05, 'epoch': 2.99}

**Comment by xyjigsaw** [comment]
你的学习率看上去已经从2e-5降到了1e-05，你的min_lr设置的是1e-05吧，另外与你的dataset数据量与step有关系。可以checkpoint一下。

**Comment by juemifuji** [comment]
> 你的学习率看上去已经从2e-5降到了1e-05，你的min_lr设置的是1e-05吧，另外与你的dataset数据量与step有关系。可以checkpoint一下。

初始学习率就是1e-5，上面我copy的有点问题
hostfile="/path/to/hostfile"
deepspeed --hostfile=$hostfile fine-tune.py
--report_to "none"
--data_path "data/belle_chat_ramdon_10k.json"
--model_name_or_path "baichuan-inc/Baichuan2-7B-Base"
--output_dir "output"
--model_max_length 512
--num_train_epochs 4
--per_device_train_batch_size 16
--gradient_accumulation_steps 1
--save_strategy epoch
--learning_rate 1e-5
--lr_scheduler_type cosine
--adam_beta1 0.9
--adam_beta2 0.98
--adam_epsilon 1e-8
--max_grad_norm 1.0
--weight_decay 1e-4
--warmup_ratio 0.0
--logging_steps 1
--gradient_checkpointing True
--deepspeed ds_config.json
--bf16 True
--tf32 True

**Comment by xyjigsaw** [comment]
要不你试试”cosine"加上双引号。

**Comment by juemifuji** [comment]
> 要不你试试”cosine"加上双引号。

确实有效。👍

**Comment by juemifuji** [comment]
需要在deepspeed配置中，加入：WarmupDecayLR
"scheduler": {
        "type": "**WarmupDecayLR**",
        "params": {
            "warmup_min_lr": "auto",
            "warmup_max_lr": "auto",
            "warmup_num_steps": "auto",
            "total_num_steps": "auto"
        }
    },
